### PR TITLE
Allow option to customize timestamp fields

### PR DESCRIPTION
**Pull request by stefankutko**
It is a requirement for us to be able to read a timestamp value from a logged message instead of relying on the global "timestamp" that appears in the log. This pull request allows for a custom start/end timestamp and defaults to the current "timestamp" if not provided.

### CODE REVIEW

**Comment by stefankutko** [comment]
Hi any update on this?  We signed the agreement.

**Comment by JuanCarniglia** [comment]
I did this same modification (without checking first if there was a PR related, my bad). I used logstash/timestamp and the method "coerce" instead of directly Time. Do you think it is better? Worse? Are you not having trouble with timezones and such, by using Datetime.parse?

**Comment by WraithM** [comment]
What's the story with this PR? This change would be super useful for me.

**Comment by WraithM** [comment]
Pinging @ph, @suyograo

**Comment by suyograo** [comment]
@mateodelnorte needs to sign the CLA since the original commits merged here are from him. See  step 2 of https://github.com/elasticsearch/logstash/blob/master/CONTRIBUTING.md#contribution-steps

**Comment by mateodelnorte** [comment]
Done. REALLY not a fan of having to complete a company-created contract in order to contribute to open source software. Your license should cover this.

**Comment by WraithM** [comment]
@suyograo, I've run this code on my logstash, it works well.

**Comment by suyograo** [comment]
@mateodelnorte thanks for completing the CLA. Unfortunately, the original commits were done using `@electronifie.com` email address, and the CLA was signed using `@gmail` address. Would you be able to update the commit with your current email? Alternatively, you could sign the CLA with `@electronifie.com` address. Thanks for your patience and contributing to Logstash

**Comment by suyograo** [comment]
@WraithM Thanks for validating this patch, but we will not be able to move forward until the CLA has been signed and matches the commits included in this patch.

**Comment by mateodelnorte** [comment]
Just out of curiosity, why does the [license](https://github.com/logstash-plugins/logstash-filter-elapsed/blob/master/LICENSE) not cover this? What additional benefit are you getting from this contract signing process?

**Comment by jordansissel** [comment]
@mateodelnorte is your question not answered by the text on this page? https://www.elastic.co/contributor-agreement (it explains a bit of why we have one, why we need it, etc)

**Comment by mateodelnorte** [comment]
TL;DR:

No, it doesn't - because it doesn't exist in a vacuum. This CLA flow is inherently less trustworthy than using a standard license, and the use of it is confusing because you have both. It would be a better use of the open source community's time to simply create or update standard license that would cover the requirements which aren't covered by those already in use.

I've gone ahead and signed the CLA because I took 30 minutes to read over the document and I'm _mostly_ sure it's ok. That said, I'd love to have that time back and simply trust the standard license in the repo. 

Reasons:

Nearly all open source projects use a standard License because 1) people don't have time or resources to have a lawyer read every legal document some company wants to draw up and 2) by using a standard License, people can trust that the contents of that license will not purposefully or unintentionally cause immediate or future harm to their persons or property. The presence of an additional contract makes me ask – Why is this not working like other projects I've contributed to?

It holds weight to keep in mind as well that even though there are standard Licenses, understanding the differences between them is so difficult that the largest open source hosting company in the world created a tool to help guide people: http://choosealicense.com/. Developers already have a hefty amount of contract knowledge to keep in mind while doing their jobs. Can you imagine an alternate reality where every company draws up their own license and redirects you to a page where you explicitly sign, instead of implicitly granting consent via participation according to the license file present and viewable as part of the codebase? 

I understand that Facebook and Google have started using this process. A far better use of everyone's time would be to create or update a standard license which can then be adopted by all. Facebook and Google are juggernauts. Everyone is going to click through their agreements without reading. As soon as every small and medium sized company starts using this process, it will cause friction to open source software development.

None of these large open source projects have a non-standard license requiring explicit signature through a flow that causes you to navigate away from the codebase to a company's privately hosted contract: https://github.com/twbs/bootstrap#contributing, https://github.com/rails/rails/blob/master/CONTRIBUTING.md, https://github.com/nodejs/node/blob/master/CONTRIBUTING.md, https://github.com/apple/swift/blob/master/CONTRIBUTING.md.

**Comment by WraithM** [comment]
@mateodelnorte, completely agreed. Thank you.

@suyograo, how can this be moved forward?

**Comment by mateodelnorte** [comment]
Why intention to merge this? It's quite old at this point. Externally hosted contacts have been signed.

**Comment by jblesener** [comment]
@suyograo How can we move this PR forward?
